171820264 error sin mando

JuanmaSalva opened the pull request:
[Finishes #171820264]
No peta pero solo reacciona a esas funcionalidades que acepten input de teclado

JuanmaSalva committed:
Peta menos

sigue petando pero puedes ver unos segundos del juego antes

JuanmaSalva committed:
Ya no peta si no tienes mando

No peta pero tampoco hace nada por que el input detecta el mando

EstebenR reviewed (changes requested):
En vez de hacer la comprobacion del tamaño siempre que se vaya a pedir el input hacerlo en la clase InputHandler. Cuando se vaya a pedir un boton asegurarse de que no se sale del numero de mandos disponibles.
Sería en los metodos de los botones, los Sticks y los triggers.

Andrés warro

eljorgemo commented:
Y si en lugar de hacer tantas comprobaciones en sitios diferentes, sólo comprobamos en el while(!exit)?
(en SDL_Game::start)

JuanmaSalva committed:
Sale excepcion

Ahora si no tienes un mando conectado te cierra la aplicacion y te sale un mensaje por consola

EstebenR reviewed (approved):
Bene, se puede cambiar una vez ya no tengamos el playstate como menu principal pero de momento funciona bien

JuanmaSalva committed:
He peusto una excepcion

He puesto una excepccion pero no sirve de nada por q sale por consola igualmente

AdriAlvarez00 commented on src/SourceFiles/SDL_Game.cpp:
```diff
@@ -93,16 +93,24 @@ void SDL_Game::closeResources() {
+	catch (int a)
+	{
+		cout << "Que tal si pones un manso eh? genio\n\n\n";
 	}
```
Si no hacemos nada en el catch creo que es mejor preguntar en los métodos de joystick si id < joystick.size()

EstebenR replied:
De acuerdo, sabiendo que no va a dar excepcion incontrolada mejor quitar el trycatch

JuanmaSalva replied:
hacerlo al principio quita un huevo y medio de comprobaciones. Si hay que hacerlo en cada método cada vez que hay input se comprueba cuando si desde el principio están todos conectados no fallará

AdriAlvarez00 reviewed (changes requested):
Hay que cambiar try catch por comprobaciones en los vectores

JuanmaSalva committed:
ya he quitado el try catch

el proximo q vuelva a pedir q lo cambie para poner lo que habia antes de este commiet se lleva una calleja.
Primer aviso <3

EstebenR committed:
Merge branch 'master' into 171820264-error-sin-mando

AdriAlvarez00 reviewed (approved):
Bien, pero hay errata en la linea 109 de SDL_Game.cpp

eljorgemo commented:
Seems alright
